renderdoc/driver/d3d8/d3d8_resources.h

```diff
@@ -0,0 +1,250 @@
+/******************************************************************************
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Baldur Karlsson
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#pragma once
+
+#include "d3d8_device.h"
+#include "d3d8_manager.h"
+
+enum ResourceType
+{
+  Resource_Unknown = 0,
+  Resource_VertexBuffer,
+  Resource_IndexBuffer,
+};
+
+ResourceType IdentifyTypeByPtr(IDirect3DResource8 *ptr);
+ResourceId GetIDForResource(IDirect3DResource8 *ptr);
+
+class TrackedResource
+{
+public:
+  TrackedResource() { m_ID = ResourceIDGen::GetNewUniqueID(); }
+  ResourceId GetResourceID() { return m_ID; }
+private:
+  TrackedResource(const TrackedResource &);
+  TrackedResource &operator=(const TrackedResource &);
+
+  ResourceId m_ID;
+};
```

Comment:
This has caused problems in the past due to the one-definition rule in C - so e.g. for D3D12 I changed it to `TrackedResource12`, here it should be `TrackedResource8`. In theory it could be moved to common code but the D3D12 variant also stores the resource record pointer, so I'm not sure that's the way to go.